### PR TITLE
mail-filter/dcc: Port to EAPI 7

### PR DESCRIPTION
**Pull request by jsmolic**
Closes: https://bugs.gentoo.org/706988

### CODE REVIEW

File: mail-filter/dcc/dcc-1.3.158.ebuild [hunk 1]
```diff
@@ -63,27 +75,29 @@ src_configure() {
...
-	DCC_SUID="${BINOWN}" DCC_OWN="${BINOWN}" DCC_GRP="${BINGRP}" export DCC_SUID DCC_OWN DCC_GRP
+	export MANOWN=root
+	export MANGRP=$(id -g -n root)
+	export BINOWN="${MANOWN}"
+	export BINGRP="${MANGRP}"
+	export DCC_PROTO_HOMEDIR="${D}/${dcc_homedir}"
+	export DCC_CGIBINDIR="${D}/${dcc_cgibin}"
+	export DCC_SUID="${BINOWN}"
+	export DCC_OWN="${BINOWN}"
+	export DCC_GRP="${BINGRP}"
```
Comment (SoapGentoo) [inline mail-filter/dcc/dcc-1.3.158.ebuild:93]: do these have to be exported?

File: mail-filter/dcc/dcc-1.3.158.ebuild [hunk 2]
```diff
@@ -63,27 +75,29 @@ src_configure() {
 moveconf() {
 	local into=/etc/dcc/
 	for i in $@; do
-		mv "${D}${dcc_homedir}/${i}" "${D}${into}"
-		dosym "${into}${i}" "${dcc_homedir}/${i}"
+		mv "${D}/${dcc_homedir}/${i}" "${D}/${into}"
```
Comment (SoapGentoo) [inline mail-filter/dcc/dcc-1.3.158.ebuild:78]: missing `|| die`

Comment (SoapGentoo) [reply mail-filter/dcc/dcc-1.3.158.ebuild:78]: You're adding `EPREFIX` everywhere, but left `${D}` everywhere?

File: mail-filter/dcc/dcc-1.3.158.ebuild [hunk 3]
```diff
@@ -14,37 +14,49 @@ SLOT="0"
...
+dcc_rundir=var/run/dcc
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-1.3.140-freebsd.patch
+	"${FILESDIR}"/${P}-fno-common.patch
+)
 
 src_configure() {
 	tc-export CC AR RANLIB
 	local myconf
```
Comment (SoapGentoo) [inline mail-filter/dcc/dcc-1.3.158.ebuild:47]: y u no use bash array?

File: mail-filter/dcc/dcc-1.3.158.ebuild [hunk 4]
```diff
@@ -14,37 +14,49 @@ SLOT="0"
 KEYWORDS="~alpha amd64 arm ~arm64 ~hppa ~ia64 ppc ppc64 sparc x86" #
```
Comment (SoapGentoo) [inline mail-filter/dcc/dcc-1.3.158.ebuild:14]: revbump and unstable, this is too invasive

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: mail-filter/dcc/dcc-1.3.158.ebuild (hunk 1 above)
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158.ebuild]: Hmm, I'm not sure :thinking:

File: mail-filter/dcc/dcc-1.3.158.ebuild (hunk 2 above)
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158.ebuild]: Done

File: mail-filter/dcc/dcc-1.3.158.ebuild (hunk 3 above)
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158.ebuild]: ya I know :D

File: mail-filter/dcc/dcc-1.3.158.ebuild (hunk 4 above)
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158.ebuild]: agreed, done

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+	# provide cronjob
+	mv "${ED}"/usr/sbin/cron-dccd "${ED}"/etc/cron.daily/dccd || die "mv failed"
+
+	# clean up
+	mv "${ED}"/usr/sbin/logger "${ED}"/usr/sbin/logger-dcc || die "mv failed"
+
+	statslist="${ED}/usr/sbin/{dcc-stats-graph,dcc-stats-init,dcc-stats-collect}"
+	if ! use rrdtool; then
+		# remove rrdtool interface scripts
+		eval rm -f ${statslist} || die "Failed to clean up rrdtool scripts"
```
Comment (SoapGentoo) [inline mail-filter/dcc/dcc-1.3.158-r1.ebuild:129]: remove the `eval` and just inline the variable

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+	fi
+
+	# clean up
+	rm -f "${ED}"/usr/sbin/{rcDCC,updatedcc}
```
Comment (SoapGentoo) [inline mail-filter/dcc/dcc-1.3.158-r1.ebuild:133]: missing `|| die`

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+	# place configuration files into /etc instead of /var/dcc
+	moveconf dcc_conf flod grey_flod grey_whitelist ids map map.txt whiteclnt whitecommon whitelist
+
+	newinitd "${FILESDIR}"/dcc.initd-1.3.154 dcc
+	newconfd "${FILESDIR}"/dcc.confd dcc
+
+	rmdir "${D}"/var/dcc/log/
```
Comment (SoapGentoo) [inline mail-filter/dcc/dcc-1.3.158-r1.ebuild:141]: ED and || die

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+		-e "s/DCCM_ARGS=\$/DCCM_ARGS='-SHELO -Smail_host -SSender -SList-ID'/;" \
+		-e "s/DCCIFD_ARGS=\$/DCCIFD_ARGS=\"\$DCCM_ARGS\"/;" \
+		-e 's/DCCIFD_ENABLE=off/DCCIFD_ENABLE=on/' \
+		-e 's/DBCLEAN_LOGDAYS=14/DBCLEAN_LOGDAYS=1/' \
+		-i "${ED}/${dcc_homedir}/dcc_conf"
+
+	if use milter ; then
+		# enable milter
+		sed -i -e "s:^[\t #]*\(DCCM_ENABLE[\t ]*=[\t ]*\).*:\1on:g" \
+			"${D}/${dcc_homedir}"/dcc_conf
```
[the remark this answers is not in the document]
Comment (SoapGentoo) [reply mail-filter/dcc/dcc-1.3.158-r1.ebuild:117]: also missign die

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+	# branding and setting reasonable defaults
+	sed -e "s/BRAND=\$/BRAND='Gentoo ${PF}'/;" \
+		-e "s/GREY_ENABLE=\$/GREY_ENABLE=off/;" \
+		-e "s/DCCM_LOG_AT=5\$/DCCM_LOG_AT=50/;" \
+		-e "s,DCCM_LOGDIR=\"log\"\$,DCCM_LOGDIR=\"/var/log/dcc\",;" \
+		-e "s/DCCM_ARGS=\$/DCCM_ARGS='-SHELO -Smail_host -SSender -SList-ID'/;" \
+		-e "s/DCCIFD_ARGS=\$/DCCIFD_ARGS=\"\$DCCM_ARGS\"/;" \
+		-e 's/DCCIFD_ENABLE=off/DCCIFD_ENABLE=on/' \
+		-e 's/DBCLEAN_LOGDAYS=14/DBCLEAN_LOGDAYS=1/' \
+		-i "${ED}/${dcc_homedir}/dcc_conf"
```
Comment (SoapGentoo) [inline mail-filter/dcc/dcc-1.3.158-r1.ebuild:112]: missing die

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+	tc-export CC AR RANLIB
+	local myconf=(
+		--bindir="${EPREFIX}"/usr/bin
+		--homedir="${EPREFIX}"/${dcc_homedir}
+		--libexecdir="${EPREFIX}"/${dcc_libexec}
+		--mandir="${EPREFIX}"/usr/share/man
+		--enable-dccifd
+		--enable-server
+		--with-DDC-MD5
+		--with-installroot=${ED}
```
[the remark this answers is not in the document]
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158-r1.ebuild]: Changed it back to `"${D}"`

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+	)
+
+	einfo "Using config: ${myconf[@]}"
+
+	# This is NOT a normal configure script.
+	./configure "${myconf[@]}" || die "configure failed!"
+}
+
+moveconf() {
+	local into=/etc/dcc/
```
[the remark this answers is not in the document]
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158-r1.ebuild]: Not really, I dropped it

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild (hunk 8 above)
[the remark this answers is not in the document]
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158-r1.ebuild]: Done

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild (hunk 5 above)
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158-r1.ebuild]: done

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild (hunk 6 above)
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158-r1.ebuild]: done

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild (hunk 7 above)
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158-r1.ebuild]: done

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild (hunk 9 above)
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158-r1.ebuild]: done

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+	)
+
+	einfo "Using config: ${myconf[@]}"
+
+	# This is NOT a normal configure script.
+	./configure "${myconf[@]}" || die "configure failed!"
+}
+
+moveconf() {
+	for i in $@; do
```
Comment (SoapGentoo) [inline mail-filter/dcc/dcc-1.3.158-r1.ebuild:76]: declare `i` as local before

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild (hunk 12 above)
Comment (jsmolic) [reply mail-filter/dcc/dcc-1.3.158-r1.ebuild:77]: done

File: mail-filter/dcc/dcc-1.3.158-r1.ebuild [hunk 13]
```diff
@@ -0,0 +1,144 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit flag-o-matic toolchain-funcs
+
+DESCRIPTION="Distributed Checksum Clearinghouse"
+HOMEPAGE="http://www.rhyolite.com/anti-spam/dcc/"
```
Comment (vaukai) [inline mail-filter/dcc/dcc-1.3.158-r1.ebuild:9]: HOMEPAGE="https://rhyolite.com/dcc/"  - maybe also for SRC_URI

**Pull request closed without merging**